### PR TITLE
Give Ruff a try

### PR DESCRIPTION
**Pull request by mirpedrol**
Add Ruff to `pyproject.toml` and as a pre-commit hook.
It can't fix rule E501 (line too long)

## PR checklist

- [ ] This comment contains a description of changes (with reason)
- [ ] `CHANGELOG.md` is updated
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] Documentation in `docs` is updated

### CODE REVIEW

**Comment by mirpedrol** [comment]
As it can't format the code, for now this would only substitute `isort`, which is probably not worth it. 
It is still a good option to add Flake and have multiple checks all in one place.
We should keep an eye on it as it's planned that it can format the code in a next version.

File: nf_core/modules/__init__.py [hunk 1]
```diff
@@ -1,15 +0,0 @@
...
-from .list import ModuleList
-from .modules_json import ModulesJson
-from .modules_repo import ModulesRepo
-from .modules_test import ModulesTest
-from .modules_utils import ModuleException
-from .mulled import MulledImageNameGenerator
-from .patch import ModulePatch
-from .remove import ModuleRemove
-from .test_yml_builder import ModulesTestYmlBuilder
-from .update import ModuleUpdate
```
Comment (mirpedrol) [inline nf_core/modules/__init__.py:15]: why are these changes in my PR?! 😱 🤯

**Comment by mirpedrol** [comment]
Update: 
Ruff does fix rule E501 (line too long), but can't fix these lines in comments or docstrings. There were 4 lines with such an error, making it unable to commit due to the pre-commit hook. 
I manually modified those lines to fix the error.
Another option will be to use the `--no-verify` flag to avoid pre-commit once we have checked that the only remained errors are E501 in comment lines.

**Comment by mashehu** [comment]
Do you know if the ignoring of comments is a tracked bug on their side or a wanted feature?

**Comment by mirpedrol** [comment]
Seems it is done on purpose. I found [this](https://stackoverflow.com/questions/70966298/python-black-code-formatter-doesnt-format-docstring-line-length) thread talking about this for black. Looks like black doesn't report it as an error, but it is not fixed.

**Comment by mirpedrol** [comment]
I have decided to remove the E501 rule as black should handle it :) 
@mashehu do you think it's fine if we maintain prettier to format files with `run_prettier_on_file`?

**Comment by mashehu** [comment]
Yes, because there shouldn't be too much overlap between the two with ruff linting code and prettier formatting it, or is there?

**Comment by mirpedrol** [comment]
I think we could replace prettier with black + ruff for python code, but for JSON for example we still need prettier.

**Comment by mashehu** [comment]
Could you please also add it to the gitpod and codespace images?

**Comment by awgymer** [comment]
> Do you know if the ignoring of comments is a tracked bug on their side or a wanted feature?

`ruff` is basically designed to work alongside `black` (which doesn't format code comments) so I think that is why it also ignores code comments. 

Are we currently not implementing any kind of `flake8` linting?

**Comment by mirpedrol** [comment]
> Are we currently not implementing any kind of `flake8` linting?

We don't lint any flake8 right now, but one good thing of adding Ruff will be to slowly lint for some of the codes that we find useful :)

**Comment by mirpedrol** [comment]
replaced by https://github.com/nf-core/tools/pull/2620

**Pull request closed without merging**